packages/neuron-wallet/src/block-sync-renderer/sync/light-connector.ts

```diff
@@ -75,30 +75,36 @@ export default class LightConnector extends Connector<CKBComponents.Hash> {
     const fetchTxHashes = fetchCellDeps
       .map(v => v.outPoint.txHash)
       .map<[string, string]>(v => ['fetchTransaction', v])
-    let txs = await this.lightRpc
-      .createBatchRequest<any, string[], (CKBComponents.TransactionWithStatus | null)[]>(fetchTxHashes)
+    const txs = await this.lightRpc
+      .createBatchRequest<any, string[], (CKBComponents.TransactionWithStatus | { status: 'fetching' | 'added' | 'not_found' })[]>(fetchTxHashes)
       .exec()
-    if (txs.some(v => !v)) {
-      // if some txs fetch to added, then fetch the actural txs
-      txs = await this.lightRpc
-        .createBatchRequest<any, string[], (CKBComponents.TransactionWithStatus | null)[]>(fetchTxHashes)
-        .exec()
+    if (txs.some(v => 'status' in v && !!v.status)) {
+      // wait for light client sync the dep cell
+      await scheduler.wait(10000)
+      return await this.getDepTxs()
     }
-    const depGroupOutputsData: string[] = fetchCellDeps
+    return fetchCellDeps
       .map((v, idx) => {
         if (v.depType === DepType.DepGroup) {
```

Comment:
`fetchCellDeps` can be filtered by `v.depType === DepType.DepGroup` before fetching transactions to reduce the payload of batch PRC and avoid this `if statement` if `map`

Reply:
It can not reduce the payload of batch RPC. If the cellDep's `depType` is not equal `DepType.DepGroup`, it also needs to fetch the cellDep's outPoint tx. And only if the cellDep's `depType` is equal `DepType.DepGroup`, it needs to fetch its dep group cell.

Reply:
I see